File: output/ChatGPT/codenet/Java/Python/s325080396.py
MAX = 200000

def main():
    R = [0] * MAX
    n = int(input())
    R = list(map(int, input().split()))
    maxv = R[1] - R[0]
    minv = R[0]
    
    for i in range(1, n):
        maxv = max(maxv, R[i] - minv)
        minv = min(minv, R[i])
    
    print(maxv)

if __name__ == "__main__":
    main()

File: output/ChatGPT/codenet/Java/C++/s325080396.cpp
#include <iostream>
using namespace std;

const int MAX = 200000;

int main() {
	int R[MAX];
	int n;
	cin >> n;
	for(int i = 0; i < n; i++) {
		cin >> R[i];
	}		
	int maxv = R[1] - R[0];
	int minv = R[0];
	for(int i = 1; i < n; i++) {
		maxv = max(maxv, R[i] - minv);	
		minv = min(minv, R[i]);
	}
	cout << maxv << endl;

	return 0;
}